R/manip_nodes.R

```diff
@@ -0,0 +1,161 @@
+#' @details
+#'   Not all functions have methods available for all object classes.
+#'   Below are the currently implemented S3 methods:
+#'  
+#'   ```{r, echo = FALSE, cache = TRUE} 
+#'   knitr::kable(available_methods(c("add_nodes", "delete_nodes", "add_node_attribute")))
+#'   ```
+#' @family modifications
+#' @inheritParams is
+#' @param attribute A named list to be added as tie or node attributes.
+#' @param attr_name Name of the new attribute in the resulting object.
+#' @param object2 A second object to copy nodes or ties from.
+#' @param vector A vector of values for the new attribute.
+#' @param ... Additional arguments.
+#' @return A data object of the same class as the function was given.
+#' @examples
+#'   other <- create_filled(4) %>% mutate(name = c("A", "B", "C", "D"))
+#'   add_nodes(other, 4, list(name = c("Matthew", "Mark", "Luke", "Tim")))
+#' @name add_nodes
+NULL
+
+#' @rdname add_nodes
+#' @param nodes The number of nodes to be added.
+#' @importFrom igraph add_vertices
+#' @export
+add_nodes <- function(.data, nodes, attribute = NULL) UseMethod("add_nodes")
+
+#' @export
+add_nodes.igraph <- function(.data, nodes, attribute = NULL){
+  igraph::add_vertices(.data, nv = nodes, attr = attribute)
+}
+
+#' @export
+add_nodes.tbl_graph <- function(.data, nodes, attribute = NULL){
+  as_tidygraph(add_nodes(as_igraph(.data), nodes, attribute))
+}
+
+#' @export
+add_nodes.network <- function(.data, nodes, attribute = NULL){
+  as_network(add_nodes(as_igraph(.data), nodes, attribute))
+}
+
+#' @rdname add_nodes
+#' @importFrom igraph delete_vertices
+#' @export
+delete_nodes <- function(.data, nodes) UseMethod("delete_nodes")
+
+#' @export
+delete_nodes.igraph <- function(.data, nodes){
+  igraph::delete_vertices(.data, v = nodes)
+}
+
+#' @rdname add_nodes 
+#' @importFrom igraph vertex_attr
+#' @export
+add_node_attribute <- function(.data, attr_name, vector){
+  if(length(vector)!=igraph::vcount(as_igraph(.data))){
+    if(is_twomode(.data) && any(length(vector) == infer_dims(.data))){
+      if(length(vector) == infer_dims(.data)[1]){
+        vector <- c(vector, rep(NA, infer_dims(.data)[2]))
+      } else if (length(vector) == infer_dims(.data)[2]){
+        vector <- c(rep(NA, infer_dims(.data)[1]), vector)
+      }
+    } else 
+      stop("Attribute vector must be same length as nodes in object.")
+  }
+  out <- as_igraph(.data)
+  igraph::vertex_attr(out, name = attr_name) <- vector
+  if(inherits(.data, "tbl_graph")) as_tidygraph(out) else
+    if(inherits(.data, "igraph")) as_igraph(out) else
+      if(inherits(.data, "igraph")) as_network(out) else
+          message("This function only works for igraph, tidygraph, or network objects.")
+}
+
+#' @rdname add_nodes
+#' @importFrom tidygraph mutate
+#' @export
+mutate_nodes <- tidygraph::mutate
+
+#' @rdname add_nodes
+#' @importFrom tidygraph mutate
+#' @export
+mutate <- tidygraph::mutate
+
+#' @export
+mutate.igraph <- function(.data, ...){
```

Comment:
This needs to be documented consistent as a method, otherwise a warning will be returned...